pkg/crypto/tinkcrypto/crypto.go

```diff
@@ -187,3 +201,123 @@ func (t *Crypto) VerifyMAC(macBytes, data []byte, kh interface{}) error {
 
 	return macPrimitive.VerifyMAC(macBytes, data)
 }
+
+// WrapKey will do ECDHES key wrapping of cek using apu, apv and recipient public key found in kh.
+func (t *Crypto) WrapKey(cek, apu, apv []byte, kh interface{}) (*composite.RecipientWrappedKey, error) {
+	// TODO: add support for 25519 key wrapping https://github.com/hyperledger/aries-framework-go/issues/1637
+	keyType := compositepb.KeyType_EC.String()
+
+	pubKH, ok := kh.(*keyset.Handle)
+	if !ok {
+		return nil, fmt.Errorf("wrapKey: %w", errBadKeyHandleFormat)
+	}
+
+	pubKey, err := keyio.ExtractPrimaryPublicKey(pubKH)
+	if err != nil {
+		return nil, fmt.Errorf("wrapKey: failed to extract recipient public key from kh: %w", err)
+	}
+
+	c, err := t.kw.getCurve(pubKey.Curve)
+	if err != nil {
+		return nil, fmt.Errorf("wrapKey: failed to get curve of recipient key: %w", err)
+	}
+
+	recPubKey := &ecdsa.PublicKey{
+		Curve: c,
+		X:     new(big.Int).SetBytes(pubKey.X),
+		Y:     new(big.Int).SetBytes(pubKey.Y),
+	}
+
+	ephemeralPriv, err := t.kw.generateKey(recPubKey.Curve)
+	if err != nil {
+		return nil, fmt.Errorf("wrapKey: failed to generate EPK: %w", err)
+	}
+
+	kek := josecipher.DeriveECDHES(ecdhesKWAlg, apu, apv, ephemeralPriv, recPubKey, keySize)
+
+	block, err := t.kw.createCipher(kek)
+	if err != nil {
+		return nil, fmt.Errorf("wrapKey: failed to create new Cipher: %w", err)
+	}
+
+	wk, err := t.kw.wrap(block, cek)
+	if err != nil {
+		return nil, fmt.Errorf("wrapKey: failed to wrap key: %w", err)
+	}
+
+	return &composite.RecipientWrappedKey{
+		KID:          pubKey.KID,
+		EncryptedCEK: wk,
+		EPK: composite.PublicKey{
+			X:     ephemeralPriv.PublicKey.X.Bytes(),
+			Y:     ephemeralPriv.PublicKey.Y.Bytes(),
+			Curve: ephemeralPriv.PublicKey.Curve.Params().Name,
+			Type:  keyType,
+		},
+		APU: apu,
+		APV: apv,
+		Alg: ecdhesKWAlg,
+	}, nil
+}
+
+// UnWrapKey unwraps a key in recWK using ECDHES with recipient private key kh.
+func (t *Crypto) UnWrapKey(recWK *composite.RecipientWrappedKey, kh interface{}) ([]byte, error) {
+	if recWK == nil {
+		return nil, fmt.Errorf("unWrapKey: RecipientWrappedKey is empty")
+	}
+
+	// only ECDHES KW alg is supported (for now)
+	if recWK.Alg != ecdhesKWAlg {
+		return nil, fmt.Errorf("unWrapKey: unsupported JWE KW Alg '%s'", recWK.Alg)
```

Comment:
Something to consider... if the caller of this method wraps the error, then you may end up with some "stuttering" in your error messages. You may get something like "failed to unwrap key: unwrapKey: unsupported JWE KW...."

I think you can omit the `unwrapKey:` part, and assume the caller will wrap it with the "unwrap key" part of the message.

Reply:
`unwrapKey` is the function name, it's helpful when troubleshooting as errors don't print the stack trace

Reply:
Yes, but the caller of the function will add "failed to unwrap key" anyways